rllib/algorithms/algorithm.py

```diff
@@ -670,6 +672,36 @@ def setup(self, config: AlgorithmConfig) -> None:
                     "offline.offline_evaluator::OfflineEvaluator"
                 )
 
+        self.trainer_runner = None
+        if self.config.get("_enable_trainer_runner", False):
+            # TODO (Kourosh/Avnishn) worker should not give us observation/action
+            # space. It should be part of the global config.
+            worker = self.workers.local_worker()
+
+            # TODO: The constructor is not clean and comprehensive.
+            self.trainer_runner = TrainerRunner(
+                trainer_class=self.config.rl_trainer_class,
```

Comment:
Let's do:

```
config = PPOConfig().resources(num_gpus=.., ).rl_module(...)

trainer_runner_config = config.get_trainer_runner_config()   # <- this is its own config object, NOT a dict!

self.trainer_runner = trainer_runner_config.build()
```

Reply:
Let's stick to one single design pattern for all major building blocks we have in RLlib (that have some complexity >= x).

Reply:
love the suggestion. Let me do it in this PR actually.